src/main/java/de/tum/in/www1/artemis/web/rest/UserJWTController.java

```diff
@@ -115,6 +133,30 @@ public ResponseEntity<JWTToken> authorizeSAML2(@RequestBody final String body) {
         return new ResponseEntity<>(new JWTToken(jwt), httpHeaders, HttpStatus.OK);
     }
 
+    @PostMapping("/personal-access-token")
+    @FeatureToggle(Feature.PersonalAccessTokens)
+    public ResponseEntity<JWTToken> getPersonalAccessToken(@RequestParam(value = "lifetimeMilliseconds", required = true) Long lifetimeMilliseconds) {
+        User user = userRepository.getUser();
+        if (!user.getActivated()) {
+            throw new UserNotActivatedException("User was disabled!");
+        }
+
+        if (lifetimeMilliseconds > this.personalAccessTokenMaxLifetimeMilliseconds)
+            return ResponseEntity.badRequest().headers(HeaderUtil.createFailureAlert(applicationName, true, ENTITY_NAME, "invalidPATLifetime",
+                    "Requested token lifetime exceeds maximum lifetime for personal access tokens!")).build();
```

Comment:
Maybe add the lifetime to the response :)

Reply:
I added the difference to the message. Adding the lifetime itself would need a custom header. The client can instead query the maximum lifetime using the GET endpoint, which is exactly what the Artemis Webapp does. This error can only occur if a custom client is used and the maximum lifetime was not restricted.

Reply:
Fine for me.